run-tests.py

```diff
@@ -105,6 +116,38 @@ def run_doctests():
         sys.exit(ret)
 
 
+def doctest_examples_readme():
+    """
+    Checks that every ipynb in the examples directory is included in the index
+    `examples/README.md`.
+    """
+    print('\nChecking that example notebooks are indexed.')
+
+    # Index file is in ./examples/README.md
+    index_file = os.path.join('examples', 'README.md')
+    assert(os.path.isfile(index_file))
+
+    with open(index_file, 'r') as f:
+        index_contents = f.read()
+
+    # Get a list of all notebooks in the examples directory
+    notebooks = [x for x in os.listdir('examples') if x.endswith('.ipynb')]
+    assert(len(notebooks) > 10)
+
+    # Find which are not indexed
+    not_indexed = [nb for nb in notebooks if nb not in index_contents]
+
+    # Report any failures
+    if len(not_indexed) > 0:
```

Comment:
Some python fans might tell you this should be `if not_indexed`; but I'm fine with this version :-)